nexus-sled-agent-shared/src/inventory.rs

```diff
@@ -223,40 +223,52 @@ impl ConfigReconcilerInventory {
             datasets,
             orphaned_datasets: IdOrdMap::new(),
             zones,
-            boot_partitions: {
-                BootPartitionContents {
-                    boot_disk: Ok(M2Slot::A),
-                    slot_a: Ok(BootPartitionDetails {
-                        header: BootImageHeader {
-                            flags: 0,
-                            data_size: 1000,
-                            image_size: 1000,
-                            target_size: 1000,
-                            sha256: [0; 32],
-                            image_name: "fake from debug_assume_success()"
-                                .to_string(),
-                        },
-                        artifact_hash: ArtifactHash([0x0a; 32]),
-                        artifact_size: 1000,
-                    }),
-                    slot_b: Ok(BootPartitionDetails {
-                        header: BootImageHeader {
-                            flags: 0,
-                            data_size: 1000,
-                            image_size: 1000,
-                            target_size: 1000,
-                            sha256: [1; 32],
-                            image_name: "fake from debug_assume_success()"
-                                .to_string(),
-                        },
-                        artifact_hash: ArtifactHash([0x0b; 32]),
-                        artifact_size: 1000,
-                    }),
-                }
-            },
+            boot_partitions: BootPartitionContents::debug_assume_success(),
             remove_mupdate_override,
         }
     }
+
+    /// Given a sled config, update an existing reconciler result to simulate an
+    /// output that sled-agent could have emitted if reconciliation succeeded.
+    ///
+    /// This method should only be used by tests and dev tools; real code should
+    /// look at the actual `last_reconciliation` value from the parent
+    /// [`Inventory`].
+    pub fn debug_update_assume_success(&mut self, config: OmicronSledConfig) {
+        let external_disks = config
```

Comment:
Is most of this basically identical to `debug_assume_success()`? I wonder if `debug_assume_success()` could call this function internally to avoid the duplication; untested but something like

```rust
let mut inv = Self {
    // these fields will be filled in by `debug_update_assume_success()`
    last_reconciled_config: OmicronSledConfig::default(),
    external_disks: BTreeMap::default(),
    datasets: BTreeMap::default(),
    orphaned_datasets: IdOrdMap::default(),
    zones: BTreeMap::default(),
    remove_mupdate_override: None,

    // these fields will not
    boot_partitions: BootPartitionContents::debug_assume_success(),
};

inv.debug_update_assume_success(config);

inv

Reply:
Updated.